samples/drivers/video/capture/src/main.c

```diff
@@ -125,10 +120,45 @@ int main(void)
 		return 0;
 	}
 
-	printk("- Default format: %c%c%c%c %ux%u\n", (char)fmt.pixelformat,
+	if (CONFIG_VIDEO_FORMAT_HEIGHT) {
+		fmt.height = CONFIG_VIDEO_FORMAT_HEIGHT;
+	}
+
+	if (CONFIG_VIDEO_FORMAT_WIDTH) {
+		fmt.width = CONFIG_VIDEO_FORMAT_WIDTH;
+		fmt.pitch = fmt.width * 2;
+	}
+
+
+#if defined(CONFIG_VIDEO_PIXEL_FORMAT_BGGR8)
+	fmt.pixelformat = VIDEO_PIX_FMT_BGGR8;
+#elif defined(CONFIG_VIDEO_PIXEL_FORMAT_GBRG8)
+	fmt.pixelformat = VIDEO_PIX_FMT_GBRG8;
+#elif defined(CONFIG_VIDEO_PIXEL_FORMAT_GRBG8)
+	fmt.pixelformat = VIDEO_PIX_FMT_GRBG8;
+#elif defined(CONFIG_VIDEO_PIXEL_FORMAT_RGGB8)
+	fmt.pixelformat = VIDEO_PIX_FMT_RGGB8;
+#elif defined(CONFIG_VIDEO_PIXEL_FORMAT_RGB565)
+	fmt.pixelformat = VIDEO_PIX_FMT_RGB565;
+#elif defined(CONFIG_VIDEO_PIXEL_FORMAT_XRGB32)
+	fmt.pixelformat = VIDEO_PIX_FMT_XRGB32;
+#elif defined(CONFIG_VIDEO_PIXEL_FORMAT_YUYV)
+	fmt.pixelformat = VIDEO_PIX_FMT_YUYV;
+#elif defined(CONFIG_VIDEO_PIXEL_FORMAT_XYUV32)
+	fmt.pixelformat = VIDEO_PIX_FMT_XYUV32;
+#elif defined(CONFIG_VIDEO_PIXEL_FORMAT_JPEG)
+	fmt.pixelformat = VIDEO_PIX_FMT_JPEG;
+#endif
```

Comment:
In case you wanted to avoid `#ifdefs` that needs to be maintained for every format, it would be possible to use a string:

For instance, this would be compiled-down step by step as:
```c
fmt.pixelformat = video_fourcc(CONFIG_VIDEO_PIXEL_FORMAT[0], CONFIG_VIDEO_PIXEL_FORMAT[1],
                               CONFIG_VIDEO_PIXEL_FORMAT[2], CONFIG_VIDEO_PIXEL_FORMAT[3]);
```
```c
fmt.pixelformat = video_fourcc("YUYV"[0], "YUYV"[1], "YUYV"[2], "YUYV"[3]);
```
```c
fmt.pixelformat = video_fourcc('Y', 'U', 'Y', 'V');
```
```c
fmt.pixelformat = 0x59555956;
```

Quick test:
```bash
echo 'int main(void) { return "\x09..."[0]; }' | cc -xc - && ./a.out; echo $?
```

Good to have more formats added to the examples!

Not sure what @loicpoulain thinks about using `video_fourcc()` directly but it helps with having new/custom formats tested without modification in the future.

Reply:
Great idea! I'll try it